fixture-monkey-tests/java-17-tests/src/test/java/com/navercorp/fixturemonkey/tests/java17/MultiLevelSealedClassTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Fixture Monkey
+ *
+ * Copyright (c) 2021-present NAVER Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.navercorp.fixturemonkey.tests.java17;
+
+import static com.navercorp.fixturemonkey.tests.TestEnvironment.TEST_COUNT;
+import static org.assertj.core.api.BDDAssertions.then;
+
+import java.beans.ConstructorProperties;
+
+import org.junit.jupiter.api.RepeatedTest;
+
+import com.navercorp.fixturemonkey.FixtureMonkey;
+import com.navercorp.fixturemonkey.api.introspector.ConstructorPropertiesArbitraryIntrospector;
+
+class MultiLevelSealedClassTest {
+	private static final FixtureMonkey SUT = FixtureMonkey.builder()
+		.objectIntrospector(ConstructorPropertiesArbitraryIntrospector.INSTANCE)
+		.defaultNotNull(true)
+		.build();
+
+	private static sealed class SealedClass permits IntermediateSealedClass {
+	}
+
+	private static sealed class IntermediateSealedClass extends SealedClass permits SealedClassImpl {
+	}
+
+	private static final class SealedClassImpl extends IntermediateSealedClass {
+		private final String value;
+
+		@ConstructorProperties("value")
+		public SealedClassImpl(String value) {
+			this.value = value;
+		}
+
+		public String getValue() {
+			return value;
+		}
+	}
```

Comment:
I think it is time to add a `SealedClassTestSpecs` for test classes.

For readability, it is better to put all classes in `SealedClassTest', `MultiLevelSealedClassTest' in `SealedClassTestSpecs'.

Reply:
Or could we just have MultiLevelSealedClassTest without SealedClassTest?